Sources/SignalProducer.swift

```diff
@@ -1329,6 +1306,12 @@ extension SignalProducer {
 	}
 }
 
+// FIXME: SWIFT_COMPILER_ISSUE
```

Comment:
For the ambiguity of `then`, my wild guess is that in concrete extensions, the compiler picks overloads with signatures with parameters resolved but no information about their original form. So the more specific overloads are ranked the same as the most generic overload.